### PR TITLE
[typing] fix typing related to `prefect.main`

### PR DESCRIPTION
**Pull request by zzstoatzz**
related to https://github.com/PrefectHQ/prefect/discussions/16167

importantly, removes `PrefectClient` and `CloudClient` from `prefect.main` since they were not included in `__all__` and therefore this is not a breaking change

### CODE REVIEW

**Review by chrisguidry — APPROVED** [review]
Thank you!

File: src/prefect/main.py [hunk 1]
```diff
@@ -67,6 +67,9 @@
     "flow",
     "Flow",
     "get_client",
+    "PrefectClient",
+    "get_cloud_client",
+    "CloudClient",
```
Comment (desertaxle) [inline src/prefect/main.py:72]: I don't think we should expose `PrefectClient` and `CloudClient` at the top level and prefer `get_client` and `get_cloud_client` instead.
```suggestion
    "get_cloud_client",
```

Comment (zzstoatzz) [reply src/prefect/main.py:72]: @desertaxle that makes sense, do you happen to know why they're imported in `prefect.main`? i.e. should I remove them from `prefect.main` entirely?

Comment (desertaxle) [reply src/prefect/main.py:72]: I'm not sure why they're imported here, but based on the fact that they weren't included in `__all__`, I think we can remove them.

Comment (zzstoatzz) [reply src/prefect/main.py]: cool, removed in [883ec70](https://github.com/PrefectHQ/prefect/pull/16288/commits/883ec7029822be4c4f37515e6903fdd01b6ad75b)

**Review by desertaxle — APPROVED** [review]

**Pull request merged**